packages/dds/tree/src/test/memory/tree.spec.ts

```diff
@@ -0,0 +1,146 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { type IMemoryTestObject, benchmarkMemory } from "@fluid-tools/benchmark";
+import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils/internal";
+import { testIdCompressor } from "../utils.js";
+
+import {
+	SchemaFactory,
+	SharedTree,
+	TreeViewConfiguration,
+	type InsertableTreeFieldFromImplicitField,
+	type TreeView,
+} from "../../index.js";
+
+const builder = new SchemaFactory("shared-tree-test");
+
+class RootNodeSchema extends builder.object("root-item", {
+	propertyOne: builder.optional(builder.number),
+	propertyTwo: builder.object("propertyTwo-item", {
+		itemOne: builder.string,
+	}),
+}) {}
+
+class EmptyRootNodeSchema extends builder.object("empty-root-item", {}) {}
+
+function createLocalSharedTree(id: string): TreeView<typeof RootNodeSchema> {
+	const sharedTree = SharedTree.create(
+		new MockFluidDataStoreRuntime({
+			registry: [SharedTree.getFactory()],
+			idCompressor: testIdCompressor,
+		}),
+		id,
+	);
+
+	const view = sharedTree.viewWith(new TreeViewConfiguration({ schema: RootNodeSchema }));
+
+	view.initialize(
+		new RootNodeSchema({
+			propertyOne: 128,
+			propertyTwo: {
+				itemOne: "",
+			},
+		}),
+	);
+
+	return view;
+}
+
+describe("SharedTree memory usage", () => {
+	// IMPORTANT: variables scoped to the test suite are a big problem for memory-profiling tests
+	// because they won't be out of scope when we garbage-collect between runs of the same test,
+	// and that will skew measurements. Tests should allocate all the memory they need using local
+	// variables scoped to the test function itself, so several iterations of a given test can
+	// measure from the same baseline (as much as possible).
+
+	beforeEach(async () => {
+		// CAREFUL: usually beforeEach/afterEach hooks are used to initialize or interact with variables
+		// whose scope is the encompasing test suite, but that's a problem for memory-profiling tests.
+		// See the comment at the top of the test suite for more details.
+	});
+
+	afterEach(() => {
+		// CAREFUL: usually beforeEach/afterEach hooks are used to initialize or interact with variables
+		// whose scope is the encompasing test suite, but that's a problem for memory-profiling tests.
+		// See the comment at the top of the test suite for more details.
+	});
+
+	benchmarkMemory(
+		new (class implements IMemoryTestObject {
+			public readonly title = "Create empty SharedTree";
+			public readonly minSampleCount = 500;
```

Comment:
I don't think we need minSampleCount here. It's useful for tests that do a lot of stuff, where the benchmark tool's heuristics for how many times can it iterate the test result in a very low number, but we need some minimum number of samples to give statistically meaningful results. I'd hope this particular test doesn't have that problem.